scilifelab/report/delivery_notes.py

```diff
@@ -206,10 +206,32 @@ def _set_project_sample_dict(project_sample_item, source):
     return project_sample_d
 
 
+def _exclude_sample_id(exclude_sample_ids, sample_name, barcode_seq):
+    """Check whether we should exclude a sample id.
+
+    :param exclude_sample_ids: dictionary of sample:barcode pairs
+    :param sample_name: project sample name
+    :param barcode_seq: the barcode sequence
+
+    :returns: True if exclude, False otherwise
+    """
+    if exclude_sample_ids and sample_name in exclude_sample_ids.keys():
+        if exclude_sample_ids[sample_name]:
```

Comment:
if "exclude_sample_ids" is None( or "", or [], or {}) the function will return None, is that okay ?
same thing if 'sample_name'  is not in exclude_sample_ids.
I'm also pretty sure you can write "sample_name in exculde_sample_ids", without adding the .keys, that should be the default behaviour of a dict. 

I'm not sure I follow the logic of this, I'll have another go later